xml/telnet_banners.xml

```diff
@@ -0,0 +1,820 @@
+    <!-- MikroTik v6.40.8 (bugfix)\r\nLogin: -->
+    <example _encoding="base64" os.version="6.40.8">TWlrcm9UaWsgdjYuNDAuOCAoYnVnZml4KQ0KTG9naW46Cg==</example>
+    <!-- MikroTik v6.36rc12 (testing)\r\nLogin: -->
+    <example _encoding="base64" os.version="6.36rc12">TWlrcm9UaWsgdjYuMzZyYzEyICh0ZXN0aW5nKQ0KTG9naW46Cg==</example>
+    <param pos="0" name="os.vendor" value="MikroTik"/>
+    <param pos="0" name="os.device" value="Router"/>
+    <param pos="0" name="os.product" value="RouterOS"/>
+    <param pos="1" name="os.version"/>
+    <param pos="0" name="os.cpe23" value="cpe:/o:mikrotik:routeros:{os.version}"/>
+    <param pos="0" name="hw.vendor" value="MikroTik"/>
+    <param pos="0" name="hw.device" value="Router"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)?ZXHN (\w+)(?: V([\d.]+))?(?:\r|\n)*Login:\s*$">
+    <description>ZTE ZXHN router</description>
+    <!-- ZXHN H108N\r\nLogin: -->
+    <example _encoding="base64" hw.product="H108N">WlhITiBIMTA4Tg0KTG9naW46Cg==</example>
+    <!-- ZXHN H298A V1.1\r\nLogin: -->
+    <example _encoding="base64" hw.product="H298A" hw.version="1.1">WlhITiBIMjk4QSBWMS4xDQpMb2dpbjoK</example>
+    <!-- ZXHN H367N\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="H367N">WlhITiBIMzY3Tg0KDUxvZ2luOgo=</example>
+    <param pos="0" name="hw.vendor" value="ZTE"/>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="0" name="hw.family" value="ZXHN"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="2" name="hw.version"/>
+  </fingerprint>
+  <fingerprint pattern="^(F6\d+\w?)\r\n\rLogin:\s*$">
+    <description>ZTE F6xx series GPON router</description>
+    <!-- F668\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="F668">RjY2OA0KDUxvZ2luOgo=</example>
+    <!-- F612W\r\n\rLogin: -->
+    <example _encoding="base64" hw.product="F612W">RjYxMlcNCg1Mb2dpbjoK</example>
+    <param pos="0" name="hw.vendor" value="ZTE"/>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="1" name="hw.product"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\r|\n)*DD-WRT v([\d.]+)(?:-(\w+))? ([\w-]+) \(c\) \d{4} NewMedia-NET GmbH(?:\r|\n)+Release: \d+\/\d+\/\d+ \(SVN revision: ([:\w]+)\)(?:\r|\n)+.* login:\s*$">
+    <description>DD-WRT - 24 family</description>
+    <!-- DD-WRT v24-sp2 mini (c) 2013 NewMedia-NET GmbH\r\nRelease: 05/27/13 (SVN revision: 21676)\r\n\r\nDD-WRT login: -->
+    <example _encoding="base64" os.version="24" os.version.version="sp2" os.edition="mini" os.build="21676">
+      REQtV1JUIHYyNC1zcDIgbWluaSAoYykgMjAxMyBOZXdNZWRpYS1ORVQgR21iSA0KUmVsZWFzZ
+      TogMDUvMjcvMTMgKFNWTiByZXZpc2lvbjogMjE2NzYpDQoNCkRELVdSVCBsb2dpbjoK
+    </example>
+    <!-- DD-WRT v24 micro (c) 2010 NewMedia-NET GmbH\r\nRelease: 08/07/10 (SVN revision: 14896)\r\n\r\nProliant DL980R07 X6550 8-core 4P SAS login: -->
+    <example _encoding="base64" os.version="24" os.edition="micro" os.build="14896">
```

Comment:
I thought this was missing a test for `os.version.version`, but maybe the point is that this example shouldn't be getting an `os.version.version` because that part of the regex is optional.  do you happen to know what happens if you added `os.version.version=""`?

Reply:
No `os.version.version` from that example.  In the past the tests fail when checking using that syntax.